frontend/src/components/filter/Filter.tsx

```diff
@@ -15,173 +14,186 @@ import {API_URL} from '../../constants';
     let verifiedurl = 'all';
-    let tagsurl = '';
-
-    status.forEach((it: any) => {
-      if (it.id === e.target.id) {
-        it.isChecked = e.target.checked;
+    let categoryurl = '';
+    const target = event.target;
+    // for handling isChecked parameter of checkbox
+    setCheckBoxStatus({...checkBoxStatus, [target.value]: target.checked});
+    status.checklist.forEach((it: any) => {
+      if (it.id === event.target.id) {
+        it.isChecked = event.target.checked;
       }
     },
     );
-    setStatus(status);
-    ddff();
-    const temptype = status.slice(0, 2);
-    const tempverified = status.slice(2, 3);
-    const temptags = status.slice(3);
-    if (tempverified[0].isChecked === true) {
+
+    const temptype = status.checklist.slice(0, 2);
+    const tempverified = status.checklist.slice(2, 3);
+    const tempcategory = status.checklist.slice(3);
+    if (tempverified[0]['isChecked'] === true) {
       verifiedurl = 'true';
     }
-    if (temptype[0].isChecked === true) {
+    if (temptype[0]['isChecked'] === true) {
       typeurl = 'task';
     }
-    if (temptype[1].isChecked === true) {
+    if (temptype[1]['isChecked'] === true) {
       typeurl = 'pipeline';
     }
-    if ((temptype[0].isChecked === true) &&
-      (temptype[1].isChecked === true)) {
+    if ((temptype[0]['isChecked'] === true) &&
+      (temptype[1]['isChecked'] === true)) {
       typeurl = 'all';
     }
-    temptags.forEach((it: any) => {
-      if (it.isChecked === true) {
-        tagsurl += it.value + '|';
+    tempcategory.map((item: any) => {
+      if (item.isChecked === true) {
+        categoryData[item.value].map((categoryitem: any) => {
+          categoryurl = categoryurl + categoryitem + '|';
+          return categoryurl;
+        });
       }
-      // if (it.isChecked === true) {
-      //   setClear('ClearAll');
-      // }
+      return tempcategory;
     });
+
+    // for displaying clear filter options
     let flag: any = false;
-    status.forEach((it: any) => {
+    status.checklist.forEach((it: any) => {
       if (it.isChecked === true) {
         flag = true;
       }
     });
     if (flag === true) {
-      setClear('ClearAll');
+      setClear('Clear All');
     } else {
       setClear(' ');
     }
-
-
-    fetch(`${API_URL}/resources/${typeurl}/${verifiedurl}?tags=${tagsurl} `)
-        .then((resp) => resp.json())
-        .then((data) => {
-          store.dispatch(
-              {
-                type: FETCH_TASK_SUCCESS,
-                payload: data,
-              },
-          );
-        });
+    fetchApi(typeurl, verifiedurl, categoryurl);
   };
 
-  // /   for clearing all checkbox
-  const cleartag = (e: any) => {
-    status.forEach((it: any) => {
-      if (it.isChecked === true) {
-        it.isChecked = false;
-      }
+  //   function for clearing all checkbox
+  const clearFilter = () => {
+    setCheckBoxStatus(
+        tempObj,
+    );
+    status.checklist.map((it: any) => {
```

Comment:
what is the point of mapping if the return value is not used?

what is the intend here?

Reply:
intend is to iterate over array ,should be foreach